src/AbstractOperations/conditional_operations.jl

```diff
@@ -86,11 +89,22 @@ function ConditionalOperation(operand::AbstractField;
     return ConditionalOperation{LX, LY, LZ}(operand, func, operand.grid, condition, mask)
 end
 
+validate_condition(cond::Function, ::ConditionalOperation) = cond
+
+function validate_condition(cond::AbstractArray, c::ConditionalOperation)
+    if ndims(cond) != 3
+        throw(ArgumentError("when kwarg condition::Array requires a 3D array of size $(size(c))"))
```

Comment:
```suggestion
        throw(ArgumentError("when kwarg condition::Array requires a 3D array of size $(size(operand))"))
```

not sure if this was intended but since this validation is in the constructor we don't have the `ConditionalOperation` yet

Reply:
oh, huh, i got confused

Reply:
true! good catch!